paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/FileStoreSourceSplitReader.java

```diff
@@ -121,6 +129,12 @@ public void handleSplitsChanges(SplitsChange<FileStoreSourceSplit> splitsChange)
         }
 
         splits.addAll(splitsChange.splits());
+        DataSplit dataSplit = (DataSplit) splits.peek().split();
+        if (currentSnapshotId == FileStoreSourceReaderMetrics.UNDEFINED
+                || dataSplit.snapshotId() != currentSnapshotId) {
+            currentSnapshotId = dataSplit.snapshotId();
+            sourceReaderMetrics.recordSnapshotUpdate(dataSplit.snapshotTimestamp());
```

Comment:
Why not use creation time of `DataFileMeta`?

```java
dataSplit.dataFiles().stream()
        .mapToLong(f -> f.creationTime().getMillisecond())
        .max()
```

In this way you don't have to add `snapshotTimestamp` into `DataSplit`, which shrinks this PR a lot.